pkgs/development/compilers/ghc/9.6.6-debian-binary.nix

```diff
@@ -0,0 +1,268 @@
+    '') extraLibraryMapping)
+
+    +
+      # Rename needed libraries and binaries, fix interpreter
+      lib.optionalString stdenv.hostPlatform.isLinux ''
+        find . -type f -executable -exec patchelf \
+            --interpreter ${stdenv.cc.bintools.dynamicLinker} {} \;
+      '';
+
+  # Not a bindist, nothing to configure
+  dontConfigure = true;
+
+  # Not a bindist, it's already built
+  dontBuild = true;
+
+  # Install prebuilt GHC files
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out
+
+    cp -t $out/ -a usr/*
+    rm -f $out/lib/ghc/lib/package.conf.d
+    find var -name "package.conf.d" -type d -exec cp -a {} $out/lib/ghc/lib/ \;
+
+    runHook postInstall
+  '';
+
+  postInstall =
+    # Patch scripts to include runtime dependencies in $PATH.
+    ''
+      for i in "$out/bin/"*; do
+        test ! -h "$i" || continue
+        isScript "$i" || continue
+        sed -i -e '2i export PATH="${lib.makeBinPath runtimeDeps}:$PATH"' "$i"
+      done
+    ''
+
+    # Patch /usr paths
+    + ''
+      for i in "$out/bin/"*; do
+        test ! -h "$i" || continue
+        isScript "$i" || continue
+        substituteInPlace "$i" \
+          --replace-fail '="/usr' '="${placeholder "out"}'
+      done
+      find "$out/lib/ghc/lib/package.conf.d" -type f -name '*.conf' \
+        -exec sed -i "s|/usr/|$out/|g" {} +
+    ''
+
+    # Patch ghc settings
+    + ''
+      substituteInPlace $out/lib/ghc/lib/settings \
+        --replace-fail powerpc64-linux-gnu-gcc gcc \
+        --replace-fail powerpc64-linux-gnu-g++ g++ \
+        --replace-fail powerpc64-linux-gnu-ld ld \
+        --replace-fail powerpc64-linux-gnu-ar ar \
+        --replace-fail powerpc64-linux-gnu-ranlib ranlib \
+        --replace-fail llc-18 llc \
+        --replace-fail opt-18 opt
+    '';
+
+  # On Linux, use patchelf to modify the executables so that they can
+  # find editline/gmp.
+  postFixup =
+    lib.optionalString (stdenv.hostPlatform.isLinux && !(debUsed.isStatic or false))
+      # Keep rpath as small as possible, running autoPatchelf makes everything segfault (maybe similar to patchelf#244).
+      # All Elfs are 2 directories deep from $out/lib, so pooling symlinks there makes a short rpath.
+      ''
+        (cd $out/lib; ln -s ${ncurses6.out}/lib/libtinfo.so.6)
+        (cd $out/lib; ln -s ${lib.getLib gmpUsed}/lib/libgmp.so.10)
+        (cd $out/lib; ln -s ${numactl.out}/lib/libnuma.so.1)
+        (cd $out/lib; ln -s ${libffi.out}/lib/libffi.so.8)
+        for p in $(find "$out/lib" -type f -name "*\.so*"); do
+          (cd $out/lib; ln -s $p)
+        done
+
+        for p in $(find "$out/lib" -type f -executable); do
+          if isELF "$p"; then
+            echo "Patchelfing $p"
+            patchelf --set-rpath "\$ORIGIN:\$ORIGIN/../.." $p
+          fi
+        done
+      ''
+    # Recache package db which needs to happen because
+    # we modify the package db
+    + ''
+      package_db=("$out"/lib/ghc*/lib/package.conf.d)
```

Comment:
I think you don't need to use a bash array here since this will always succeed. This was just required for some bindists that sometimes used the hadrian and sometimes the old make dir layout depending on platform.